sql_generators/glean_usage/events_stream.py

```diff
@@ -126,4 +148,72 @@ def generate_per_app(
             parallelism=parallelism,
             id_token=id_token,
             all_base_tables_exist=all_base_tables_exist,
+            custom_render_kwargs=custom_render_kwargs,
         )
+
+
+@cache
+def get_glean_app_repository(v1_name: str) -> dict:
+    """Return the Glean app's repository."""
+    for repository in get_glean_repositories():
+        if repository["name"] == v1_name:
+            return repository
+    raise Exception(f"No Glean repository found for app `{v1_name}`.")
+
+
+@cache
+def get_glean_dependency_v1_name(dependency: str) -> str:
+    """Return the v1 name of the Glean dependency, which may be a library."""
+    for repository in get_glean_repositories():
+        if "library_names" in repository and dependency in repository["library_names"]:
+            return repository["name"]
+    return dependency
+
+
+@cache
+def get_glean_app_event_extras_by_type(
+    v1_name: str, ping: str = "events", cutoff_date: Optional[date] = None
+) -> dict[str, set[str]]:
+    """Return the Glean app's event extra keys for the specified ping, grouped by type."""
+    extras_by_type = defaultdict(set)
+    repository = get_glean_app_repository(v1_name)
+
+    if not cutoff_date:
+        ping_delete_after_days = (
```

Comment:
Just fyi, it's possible that one day we will support setting retention in dependencies so the way it's using the passed in date may longer be relevant: DENG-7765.  But nothing to do now and the dependency would likely be set lower than the app so using an earlier cutoff date is fine